### PR TITLE
Tracer startup benchmark events

### PR DESCRIPTION
**Pull request by ValentinZakharov**
# What Does This Do
Added events logging into CSV to track initialisation time.

# Motivation

# Additional Notes
This changes is a part of initiative to integrate microbenchmarking platform with java tracer

### CODE REVIEW

**Comment by richardstartin** [comment]
Is this something you could achieve with less overhead with logging to stdout? JFR needs to instrument the methods timed here, which takes time, involves bootstrapping JFR and so on. Care has been taken in the past to avoid initialising JFR this early.

**Comment by mcculls** [comment]
I was going to suggest the same as @richardstartin - I would prefer to see a logging-based solution during premain

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/benchmark/StaticEventLogger.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.Objects;
+
+public class StaticEventLogger {
+
+    private static final Logger log = LoggerFactory.getLogger(StaticEventLogger.class);
+
+    private static final int EVENT_BEGIN = '1';
+    private static final int EVENT_END = '0';
+
+    private static BufferedWriter out = null;
```
Comment (mcculls) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/benchmark/StaticEventLogger.java:17]: if you can make this `static final` then the various calls could be removed by the JIT when this is off

one way to do that would be to check the property inside a static initializer in this class

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/benchmark/StaticEventLogger.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    public static synchronized void startLogging(String fileName) {
```
Comment (mcculls) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/benchmark/StaticEventLogger.java:19]: agent startup is single-threaded, so `synchronized` is not necessary here

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java [hunk 3]
```diff
@@ -121,6 +123,10 @@ public boolean isEnabledByDefault() {
   private static boolean debuggerEnabled = false;
 
   public static void start(final Instrumentation inst, final URL agentJarURL) {
+    StaticEventLogger.begin("Tracer");
+    StaticEventLogger.begin("Agent.start");
+
+    createAgentClassloader(agentJarURL);
```
Comment (mcculls) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java:129]: remove this duplicate call - it already exists on the line below :)

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java [hunk 4]
```diff
@@ -121,6 +123,10 @@ public boolean isEnabledByDefault() {
   private static boolean debuggerEnabled = false;
 
   public static void start(final Instrumentation inst, final URL agentJarURL) {
+    StaticEventLogger.begin("Tracer");
```
Comment (mcculls) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java:126]: This event isn't strictly correct - the tracer part of the agent install happens later...

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java (hunk 3 above)
Comment (ValentinZakharov) [reply dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java:128]: Fixed

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java (hunk 4 above)
Comment (ValentinZakharov) [reply dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java]: Thanks, renamed to `Agent`

**Review by mcculls — APPROVED** [review]

**Review by ygree — APPROVED** [review]
LGTM

**Pull request merged**